src/os/bluestore/BlueStore.cc

```diff
@@ -9873,8 +9856,11 @@ int BlueStore::queue_transactions(
   for (auto c : on_applied_sync) {
     c->complete(0);
   }
-  for (auto c : on_applied) {
-    finishers[osr->shard]->queue(c);
+  if (on_applied.size()) {
```

Comment:
nit, might want to put `if (!on_applied.empty())` if we don't care about the "size" here.